Archivos.c

```diff
@@ -0,0 +1,66 @@
+#include"Archivos.h"
+
+/*1)*/Archivos::Archivos(){
+    //ctor sin patamentros
+        abierto=false;
+
+}
+
+/*2)*/Archivos::Archivos(string camino){
+    //ctor
+    abrirArchivo(string camino);
+    if(this->archivo.good()){
+        abirto=true;
+    }else{
+        abierto=false;
+    }
+
+}
+
+/*3)*/Archivos::~Archivos(){
+    //dtor
+    if(this->estaAbierto()){
+        this->cerrarArchivo();
+    }
+}
+
+/*4)*/void Archivos::abrirArchivo(string camino){
+
+    try{
+        archivo.open(camino);
+        abierto=true;
+        if(this->estaAbierto()){
+            throw abierto;
+        }
+        if(archivo.fail()){
+            abierto=false;
+            throw abierto;
+        }
```

Comment:
Ojo aca porque siempre se va a lanzar la excepcion. La idea seria lanzarla si el archivo ya estaba abierto, o si no se pudo abrir. Para eso el orden seria
1. Fijarse si el archivo esta abierto, en ese caso lanzar la excepcion (y ahi sale del try)
2. Abrir el archivo y poner abierto en true
3. Si el archivo no se pudo abrir, poner abierto en false y lanzar la excepcion
```suggestion
    try{
        if(this->abierto){
            throw abierto;
        }
        archivo.open(camino);
        abierto=true;
        if(archivo.fail()){
            abierto=false;
            throw abierto;
        }
```